File: app/src/main/java/com/group15/fitnesstracker/db/dao/TrainerDao.java
package com.group15.fitnesstracker.db.dao;

import com.group15.fitnesstracker.db.Trainer;

import java.util.List;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import io.reactivex.Completable;
import io.reactivex.Maybe;

public interface TrainerDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public Completable insertTrainers(Trainer... Trainers);

    @Update
    public Completable updateTrainers(Trainer... Trainers);

    @Delete
    public Completable deleteTrainers(Trainer... Trainers);

    @Query("SELECT * FROM Trainer")
    public Maybe<List<Trainer>> loadAllTrainers();
}

File: app/src/main/java/com/group15/fitnesstracker/db/dao/TrainerDao.kt
package com.group15.fitnesstracker.db.dao

import androidx.room.*
import com.group15.fitnesstracker.db.Trainer

import io.reactivex.Completable
import io.reactivex.Maybe

@Dao
interface TrainerDao {
    @Insert
    fun createTrainer(trainer: Trainer): Maybe<Long>

    @Update
    fun updateTrainers(vararg Trainers: Trainer): Completable

    @Delete
    fun deleteTrainers(vararg Trainers: Trainer): Completable

    @Query("SELECT * FROM Trainer")
    fun loadAllTrainers(): Maybe<List<Trainer>>

    @Query("select * from Trainer where id = :id")
    fun getTrainer(id: Int): Maybe<Trainer>
}
